elastic/kibana
# [Index Management] Index details page: loading index data

## Summary
Follow up to https://github.com/elastic/kibana/pull/163521

- A new internal api endpoint to load index data for a single index specified by the index name
- An api service to send a request to the new endpoint
- A loading and an error state on the new index details page
- A "back to all indices" button
- A button with the link to Discover
- A context menu with index actions (only popover with options is displayed, but the actions are not implemented yet)

### Screenshots 

<img width="1143" alt="Screenshot 2023-08-23 at 14 31 26" src="https://github.com/elastic/kibana/assets/6585477/06c3eea5-8d35-4c09-9274-fe545bde60ae">


<img width="740" alt="Screenshot 2023-08-22 at 17 43 13" src="https://github.com/elastic/kibana/assets/6585477/11433e81-ef21-43b8-83b4-9c6b4c1c7579">

<img width="644" alt="Screenshot 2023-08-22 at 17 42 30" src="https://github.com/elastic/kibana/assets/6585477/f6649029-af31-47d2-ba6d-4632c0c5b146">


### How to test
1. Add `xpack.index_management.dev.enableIndexDetailsPage: true` to your `/config/kibana.dev.yml` file 
2. Start ES and Kibana with `yarn es snapshot` and `yarn start`
3. Add at least 1 sample data test
4. Navigate to Index Management and click the name of any index
5. Check that the button "back to indices" works (navigates back to the indices list)
6. Check that the button "discover index" works (navigates to Discover)
7. Check that the button "manage index" opens the context menu (_note: index actions are not implemented yet_)
8. Check that there is a loading indicator (by adding throttle in the Network tab of Chrome Dev tools)
9. Check that the error section is displayed and the "reload" button resends a request (by blocking the request in the Network tab of Chrome Dev Tools)

### Checklist
